### DISCUSSION TITLE
Warn about NaNs when interpolating?

### DISCUSSION THREAD

**Discussion by johnomotani** [pydata/xarray#5842] [Ideas]
When using `DataArray.interp()` to interpolate a 3d array along one dimension, I was surprised to end up with an output full of NaN. There were NaNs in a few elements in one corner of the original array, but the result has NaN even at points where the interpolation does not depend on the NaN in the input. I think the interpolation is done by `scipy.interpolate.interp1d`, whose [documentation says](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html):

> Calling interp1d with NaNs present in input values results in undefined behaviour.

Probably we can't do anything to actually fix this, but I think it'd be nice to at least raise a warning or an error if the input has any NaNs, so that the user can do something about it (possibly suggest using `DataArray/Dataset.fillna`?).

**Comment by TomNicholas** [comment]
That definitely seems like something we should warn about! Perhaps even raise an error. Could you raise an issue with a minimal example? Also do you know what happens if you feed `scipy.interpolate.interpn` any NaNs?
